pkg/sdk/tasks_impl_gen.go

```diff
@@ -60,6 +63,45 @@ func (v *tasks) Execute(ctx context.Context, request *ExecuteTaskRequest) error
 	return validateAndExec(v.client, ctx, opts)
 }
 
+// GetRootTasks is a way to get all root tasks for the given tasks.
+// Snowflake does not have (yet) a method to do it without traversing the task graph manually.
+// Task DAG should have a single root but this is checked when the root task is being resumed; that's why we return here multiple roots.
+// Cycles should not be possible in a task DAG but it is checked when the root task is being resumed; that's why this method has to be cycle-proof.
+// TODO [SNOW-884987]: handle cycles
+func GetRootTasks(v Tasks, ctx context.Context, id SchemaObjectIdentifier) ([]Task, error) {
+	task, err := v.ShowByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	predecessors := task.Predecessors
+	// no predecessors mean this is a root task
+	if len(predecessors) == 0 {
+		return []Task{*task}, nil
+	}
+
+	rootTasks := make([]Task, 0, len(predecessors))
+	for _, predecessor := range predecessors {
+		predecessorTasks, err := GetRootTasks(v, ctx, predecessor)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get predecessors for task %s err = %w", predecessor.FullyQualifiedName(), err)
+		}
+		rootTasks = append(rootTasks, predecessorTasks...)
+	}
+
+	// TODO [SNOW-884987]: extract unique function in our collection helper (if cycle-proof algorithm still needs it)
+	keys := make(map[string]bool)
+	uniqueRootTasks := make([]Task, 0, len(rootTasks))
+	for _, rootTask := range rootTasks {
+		if _, value := keys[rootTask.ID().FullyQualifiedName()]; !value {
```

Comment:
could you change that to `exists` or `ok` ? I was confused for sec, because the first returned value is the actual map value and the second one is existence check called "value"

Reply:
Done